src/util.rs

```diff
@@ -522,6 +527,43 @@ impl Run {
                     Some(Interpretable::Bril(bril)),
                 )
             }
+            RunType::Compile => {
+                let opt_level = "none"; // options are "none", "speed", and "speed_and_size"
+                let object = self.name() + ".o";
+                brilift::compile(
+                    &self.prog_with_args.program,
+                    None,
+                    &object,
+                    opt_level,
+                    false,
+                );
+
+                let executable = self.name();
+                std::process::Command::new("cc")
+                    .arg(object.clone())
+                    .arg("infra/brilift.o")
+                    .arg("-o")
+                    .arg(executable.clone())
+                    .status()
+                    .unwrap();
+
+                let executable = std::path::Path::new(&executable).canonicalize().unwrap();
+
+                std::process::Command::new("rm")
+                    .arg(object)
+                    .status()
+                    .unwrap();
+
+                // TODO: only do this during tests
+                if !self.interp {
```

Comment:
We should probably produce temporary files and directories for tests
Here's a crate for that: https://docs.rs/tempdir/latest/tempdir/